tests/queries/0_stateless/02834_timestamp_function.sql

```diff
@@ -0,0 +1,5 @@
+SELECT TIMESTAMP('2003-12-31');
```

Comment:
MySQL defines `TIMESTAMP` as a data type. In terms of bounds, the natural equivalent in CH would be `DateTime`. However, `DateTime` does not have a fractional part (microsecond resolution), so in order to accept all inputs it would need to be DateTime64 but with fixed precision.

So the options are either: Implement TIMESTAMP as a function (like in this PR) or as a data type alias. The latter would be more natural and give conversion from strings and other types (e.g. integers) to TIMESTAMP for free. You probably want to look into this direction, see e.g. how IPv4/6 are implemented.

Reply:
> it would need to be DateTime64 but with fixed precision.
Yes, DateTime64(6)

> Implement TIMESTAMP as a function (like in this PR) or as a data type alias
We would need to implement a separate function for it either way because we have a second argument - the time to add to the first argument after parsing.